File: solutions/csharp/diamond/1/Diamond.cs
public static class Diamond
{
    public static string Make(char target)
    {
        if (target < 'A' || target > 'Z')
            throw new ArgumentOutOfRangeException(nameof(target), "Target must be an uppercase letter from A to Z.");
        var size = 2*(target - 'A') + 1;
        var rows = new string[size];
        for (char c = 'A'; c <= target; c++)
        {
            var row = new char[size];
            Array.Fill(row, ' ');
            int index = c - 'A';
            row[size / 2 - index] = c;
            row[size / 2 + index] = c;
            rows[index] = new string(row);
            rows[size - 1 - index] = new string(row);
        }
        return string.Join("\n", rows);
    }
}

File: solutions/go/diamond/1/diamond.go
package diamond

import (
	"bytes"
	"errors"
	"strings"
)

func Gen(char byte) (string, error) {
	if char < 'A' || char > 'Z' {
		return "", errors.New("char out of range")
	}
	rowLth := 2*(char-'A') + 1
	rows := make([]string, rowLth)
	for c := byte('A'); c <= char; c++ {
		row := bytes.Repeat([]byte{' '}, int(rowLth))
		colNum := char - c
		row[colNum] = c
		row[rowLth-colNum-1] = c
		rowNum := c - 'A'
		rows[rowNum] = string(row)
		rows[rowLth-rowNum-1] = string(row)
	}
	return strings.Join(rows, "\n"), nil
}
